### ISSUE TITLE
false positives of bugprone-sizeof-expression with a plain template type argument

### ISSUE THREAD

**Issue by sfc-gh-sgiesecke** [llvm/llvm-project#140240]
When a template like
```
template <typename T>
void unalignedStore(void* const dst, const T& src) noexcept {
  std::memcpy(dst, &src, sizeof(T));
}
```
is instantiated with `Foo*` such that `Foo` is an aggregate, then this produces a diagnostic of `bugprone-sizeof-expression`:
```
error: suspicious usage of 'sizeof(A*)' on pointer-to-aggregate type; did you mean 'sizeof(A)'?
```

I'd say that it generally makes sense to emit a warning if the sizeof expression is of the form  `sizeof(T)` where `T` is a template type argument.

In addition, even if it makes sense, the diagnostic is not too helpful, as it doesn't mention the context of instantiation or the actual template type argument. `A*/A` is just a placeholder, which is generally confusing.

**Comment by voyager-jhk** [comment]
I tried reproducing this on the latest main branch, but the issue no longer appears.

The checker still correctly warns on direct expressions like sizeof(ptr), but it does not emit warnings for the template cases shown here (e.g., unalignedStore or load).

This is likely related to recent infrastructure changes where matchers default to TK_IgnoreUnlessSpelledInSource, which prevents traversal into implicitly instantiated template code.

Reproduction attempt on recent trunk:

```bash
$ cat test.cpp
struct Foo { int a; };
void* memcpy(void* dest, const void* src, unsigned long count);

template <typename T> 
void unalignedStore(void* dst, const T& src) {
    memcpy(dst, &src, sizeof(T));
}

void test_issue() {
    Foo* p = nullptr;
    void* buffer = nullptr;
    unalignedStore(buffer, p);
}

$ clang-tidy -checks='-*,bugprone-sizeof-expression' test.cpp --
(No warnings emitted)
```

Since the issue is no longer reproducible, it seems reasonable to consider this resolved.
